pkg/nmi/server/endpointpolicy.go

```diff
@@ -85,31 +85,45 @@ func DeleteEndpointRoutePolicy(podIP string, metadataIP string) error {
 }
 
 func getEndpointByIP(ip string) (*v1.HNSEndpoint, error) {
-	klog.Infof("Getting endpoint for IP %s\n", ip)
-
 	request := msg.HNSRequest{
 		Entity:    msg.EndpointV1,
 		Operation: msg.Enumerate,
 		Request:   nil,
 	}
 
-	klog.Info("Enumerating all endpoints\n")
-	response, err := callHcnProxyAgent(request)
-	if err != nil {
-		return nil, &endpointPolicyError{InvalidOperation, err}
-	}
+	retryCount := 1
+	maxRetryCount := 4
+	var sleepFactor time.Duration = 1
 
-	var endpoints []v1.HNSEndpoint
-	err = json.Unmarshal(response, &endpoints)
-	if err != nil {
-		return nil, &endpointPolicyError{InvalidOperation, err}
-	}
+	for {
+		klog.Infof("Getting endpoint for IP %s\n", ip)
+		response, err := callHcnProxyAgent(request)
```

Comment:
If this function return error, it won't retry. Because callHcnProxyAgent() itself has retry logic inside. If we retry when it still fails, it would cost too much time